### PR TITLE
Refactor filters, add columns on attendee list

### PR DESCRIPTION
**Pull request by tomrndom**
image

ref: https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskProject&view=vertical&task=3325512&stafilter=NotComplete&fileview=grid

Signed-off-by: Tomás Castillo <tcastilloboireau@gmail.com>

### CODE REVIEW

File: src/actions/attendee-actions.js [hunk 1]
```diff
@@ -133,20 +129,38 @@ const parseFilters = (filters) => {
           ''
         ));
     }
+
+    if (filters.checkinDateFilter && filters.checkinDateFilter.some(e => e !== null)) {
+        console.log('a1asdalsdjkhaskjbk1')
```
Comment (smarcet) [inline src/actions/attendee-actions.js:134]: remove this

File: src/actions/attendee-actions.js [hunk 2]
```diff
@@ -193,6 +224,23 @@ export const exportAttendees = (order = 'id', orderDir = 1, filters = {}) => asy
 
     const filter = parseFilters(filters);
 
+    if (term) {
```
Comment (smarcet) [inline src/actions/attendee-actions.js:227]: could we refactor this and have this parsing logic on a single place ?

File: src/actions/attendee-actions.js [hunk 3]
```diff
@@ -464,6 +513,23 @@ export const sendEmails = (currentFlowEvent,
...
```
Comment (smarcet) [inline src/actions/attendee-actions.js:516]: save as above move to parseFilters

File: src/actions/attendee-actions.js [hunk 4]
```diff
@@ -518,4 +584,8 @@ const normalizeEntity = (entity) => {
     }
 
     return normalizedEntity;
-};
\ No newline at end of file
+};
+
+export const changeAttendeeListSearchTerm = (term) => (dispatch, getState) => {
```
Comment (smarcet) [inline src/actions/attendee-actions.js:589]: this method is not used ?

**Review by smarcet — CHANGES_REQUESTED** [review]
@tomrndom please review comments

File: src/actions/attendee-actions.js (hunk 4 above)
Comment (tomrndom) [reply src/actions/attendee-actions.js:554]: this is used at https://github.com/fntechgit/summit-admin/blob/f7761db9d75c63cbed5d170cf0b2df62cb4f9128/src/pages/attendees/summit-attendees-list-page.js#L325

File: src/pages/attendees/summit-attendees-list-page.js [hunk 5]
```diff
@@ -50,164 +59,65 @@ class SummitAttendeeListPage extends React.Component {
...
+        this.handleExtraFilterChange = this.handleExtraFilterChange.bind(this);
+        this.handleApplyEventFilters = this.handleApplyEventFilters.bind(this);
+        this.handleTermChange = this.handleTermChange.bind(this);
         this.state = {
             showModal: false,
             modalTitle: '',
-            modalSchedule: []
+            modalSchedule: [],
+            enabledFilters: [],
+            eventFilters: {
```
Comment (smarcet) [inline src/pages/attendees/summit-attendees-list-page.js:81]: this should be renamed properly

File: src/pages/attendees/summit-attendees-list-page.js [hunk 6]
```diff
@@ -50,164 +59,65 @@ class SummitAttendeeListPage extends React.Component {
...
+                memberFilter: null,
+                statusFilter: null,
+                ticketsFilter: null,
```
Comment (smarcet) [inline src/pages/attendees/summit-attendees-list-page.js:84]: default is   ticketsFilter: 'HAS_TICKETS',

File: src/pages/attendees/summit-attendees-list-page.js [hunk 7]
```diff
@@ -28,11 +29,19 @@ import {
...
+    changeAttendeeListSearchTerm
 } from "../../actions/attendee-actions";
 
 import {getBadgeFeatures, getBadgeTypes} from "../../actions/badge-actions";
 
+const fieldNames = [    
+    { columnKey: 'member_id', value: 'member_id', sortable: true},
+    { columnKey: 'tickets_qty', value: 'tickets_qty' },
+    { columnKey: 'company', value: 'company', sortable: true},
+    { columnKey: 'checkin_date', value: 'checkin_date'},
```
Comment (smarcet) [inline src/pages/attendees/summit-attendees-list-page.js:42]: this should be sorteable

**Review by smarcet — CHANGES_REQUESTED** [review]
@tomrndom please review comments

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**